ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/TlsConfiguration.scala

```diff
@@ -3,70 +3,166 @@
-      )
-    else None
+  def client(enabledProtocols: Seq[TlsVersion] = Seq.empty): Option[SslContext] = {
+    if (enabled) {
+      val enabledProtocolsNames =
+        if (enabledProtocols.isEmpty)
+          null
+        else
+          enabledProtocols.map(_.version).asJava
+      val sslContext = GrpcSslContexts
+        .forClient()
+        .keyManager(
+          keyCertChainFile.orNull,
+          keyFile.orNull,
+        )
+        .trustManager(trustCertCollectionFile.orNull)
+        .protocols(enabledProtocolsNames)
+        .sslProvider(SslContext.defaultClientProvider())
+        .build()
+      logTlsProtocolsAndCipherSuites(sslContext, isServer = false)
+      Some(sslContext)
+    } else None
   }
 
   /** If enabled and all required fields are present, it returns an SslContext suitable for server usage */
   def server: Option[SslContext] =
     if (enabled) {
-      scala.util.Using.resources(
+
+      val tlsInfo = scala.util.Using.resources(
         keyCertChainInputStreamOrFail,
         keyInputStreamOrFail,
       ) { (keyCertChain: InputStream, key: InputStream) =>
-        Some(
-          GrpcSslContexts
-            .forServer(
-              keyCertChain,
-              key,
-            )
-            .trustManager(trustCertCollectionFile.orNull)
-            .clientAuth(clientAuth)
-            .protocols(if (protocols.nonEmpty) protocols.asJava else null)
-            .build
+        val protocols = null.asInstanceOf[lang.Iterable[String]]
+        val defaultSslContext = buildServersSslContext(
+          keyCertChain = keyCertChain,
+          key = key,
+          protocols = protocols,
         )
+        TlsInfo.fromSslContext(defaultSslContext)
+      }
 
+      scala.util.Using.resources(
+        keyCertChainInputStreamOrFail,
+        keyInputStreamOrFail,
+      ) { (keyCertChain: InputStream, key: InputStream) =>
+        val protocols = protocolsNames(tlsInfo)
+        val sslContext = buildServersSslContext(
+          keyCertChain = keyCertChain,
+          key = key,
+          protocols = protocols,
+        )
+        logTlsProtocolsAndCipherSuites(sslContext, isServer = true)
+        Some(sslContext)
       }
-    } else None
+
+    } else {
+      logger.info(s"Server's TLS: Disabled.")
+      None
+    }
+
+  private def buildServersSslContext(
+      keyCertChain: InputStream,
+      key: InputStream,
+      protocols: lang.Iterable[String],
+  ) = {
+    GrpcSslContexts
+      .forServer(
+        keyCertChain,
+        key,
+      )
+      .trustManager(trustCertCollectionFile.orNull)
+      .clientAuth(clientAuth)
+      .protocols(protocols)
+      .sslProvider(SslContext.defaultServerProvider())
+      .build()
+  }
+
+  private[tls] def logTlsProtocolsAndCipherSuites(
+      sslContext: SslContext,
+      isServer: Boolean,
+  ): Unit = {
+    val who = if (isServer) "server" else "client"
+    val tlsInfo = TlsInfo.fromSslContext(sslContext)
+    logger.info(s"TLS $who: Enabled.")
```

Comment:
I would rephrase this so that we say things such as:
Server TLS - enabled
Server TLS - supported protocols: <prot>, <prot>, <prot>

Reply:
Ok